### PR TITLE
feat: add loading state to button

### PR DESCRIPTION
**Pull request by rendyandika28**
Hi creator,

I hope you're well! I've created this pull request to add a new feature to the Button component to have a loading state.

Changes:
- Add loading props in the button component
- Add a slot for the loader icon, so it can be more customized

Testing:
I've performed thorough testing to ensure the new functionality works as expected:
- Unit tests for the loader icon

Notes:
I've updated the codebase documentation and comments to reflect the changes accurately.

Please review this pull request when you have time. Your feedback is valuable, and I'm open to making any necessary adjustments.

Thank you!

Best regards,
Rendy Andika

### CODE REVIEW

File: src/components/button/index.md [hunk 1]
```diff
@@ -5,7 +5,19 @@ description: Base button component
 
 <script setup>
   import pButton from './Button.vue'
+  import pSpinner from '../spinner/spinner.vue'
```
Comment (adenvt) [inline src/components/button/index.md:8]: Build failed

It should be written in Capital case, `../spinner/Spinner.vue`.

**Review by adenvt — CHANGES_REQUESTED** [review]
I need to consult with UI/UX first for this.

Because at now, there no guideline for loading state on button.
It will make confusion where spinner should be placed, before or after the button content. We need rule for this.

File: src/components/button/index.md (hunk 1 above)
Comment (rendyandika28) [reply src/components/button/index.md]: done

**Comment by adenvt** [comment]
@rendyandika28 After discuss with Designer, I'm find out they already have guide for loading state on button

It look like this

image

Or see [live preview here](https://www.figma.com/proto/TFfoZRtIffRlzHm9R2EvbB/B-A-S-E-%2F-Mobile---Style-Guide?node-id=1015-86467&scaling=min-zoom&page-id=262%3A2158)

Do you mind to change like that? I drop figma link in [here](https://www.figma.com/file/TFfoZRtIffRlzHm9R2EvbB/B-A-S-E-%2F-Mobile---Style-Guide?node-id=1015%3A86467&mode=dev)

**Comment by adenvt** [comment]
Duplicate #791

**Pull request closed without merging**